modules/gdscript/editor/script_templates/CompositorEffect/post_process_shader.gd

```diff
@@ -0,0 +1,149 @@
+@tool
+extends CompositorEffect
+class_name PostProcessShader
+
+const template_shader: String = "#version 450
+
+// Invocations in the (x, y, z) dimension
+layout(local_size_x = 8, local_size_y = 8, local_size_z = 1) in;
+
+layout(rgba16f, set = 0, binding = 0) uniform image2D color_image;
+
+// Our push constant
+layout(push_constant, std430) uniform Params {
+	vec2 raster_size;
+	vec2 reserved;
+} params;
+
+// The code we want to execute in each invocation
+void main() {
+	ivec2 uv = ivec2(gl_GlobalInvocationID.xy);
+	ivec2 size = ivec2(params.raster_size);
+
+	if (uv.x >= size.x || uv.y >= size.y) {
+		return;
+	}
+
+	vec4 color = imageLoad(color_image, uv);
+
+	#COMPUTE_CODE
+
+	imageStore(color_image, uv, color);
+}"
+
+@export_multiline var shader_code: String = "":
+	set(value):
+		mutex.lock()
+		shader_code = value
+		shader_is_dirty = true
+		mutex.unlock()
+
+var rd: RenderingDevice
```

Comment:
This template contains new type hints that should be stripped here:

https://github.com/godotengine/godot/blob/71699e08c9df78b7203fa4ef9cede28e995d6ace/modules/gdscript/gdscript_editor.cpp#L81-L97